### PR TITLE
refactor: update-nx-e2e-tests-builds

### PR DESCRIPTION
**Pull request by ryanbas21**
# Description
Dependent on #422 

There is a lot in this PR so i'm going to try to break it down.

Core stuff is an NX upgrade to 18. With that comes a lot of features. One is that NX now will be able to shard playwright tests. That means we can run our autoscript suites in a more granular way, and if something is for some reason flakey, we can mark it as so in the NX web app. NX can then re-run flakey tests for us if they are flakey.

I did a dependency clean up. Unified all our build tooling so that it is as follows:

- packages/* will use tsc (this enables us to use nx release which is another nx 18ish feature).
- e2e/* will all use vite for builds and playwright for testing.
- removed old scripts from the repo. Updated husky scripts to run commands.
- cleaned up the package.json scripts.
- Removed all of the dependencies from rollup (some are needed for vite) that existed in packages/*/package.json
- Removed samples in favor of new repo move coming up. Also removed all dependenices related to that.
- rewrote CI to the new nx agent style which is faster and simplier.
- rewrote all tests from jest to vitest. removed jest.
- refactored autoscript-suites to use local host (mock api as well). This means we dont have fake certs and dont have to manage domains in CI or locally.
- removed old devcontainer
- its important that a developer has NX_ADD_PLUGINS=false so i put that in the .env.example file. Its easier to just put that in the .zshrc or bash file but its there in case someone needs it when cloning the repo. This enables a feature of NX called project crystal which does the sharding mentioned above and more.
- fixed the token-vault-app/proxy and updated token-vault-suites to make tests work again
- updated a tools/releases/local.mjs file which will spin up verdaccio, and then run nx release on a beta tag and prerelease id and publish all `packages/*` sync'd to the same tag/version to verdaccio.

The goal here is reduce complexity, update everything as we were behind, and stabilize our tests. It appears as of now that goal is achieved and we can move closer to fully automating releasing and more.

Technically this PR can go in as it is, and leave the other PR's alone, I broke it out slowly but this PR ended up getting most of the work. This will be demo'd so its easier to digest.

How can you test this?

Use this branch locally, familiarize yourself with the changes. it shouldn't be drastically different than what we had before.

in your `~/.npmrc` you can add a line like so `//localhost:4873/:_authToken=dummy_token`

in your .zshrc you can add `NX_ADD_PLUGINS=false` or we can keep a shared .env file in the repo with this variable.

`npm run build` will run build on all projects
`npm run e2e` will run e2e on all e2e apps

`npm run lint` will lint all projects
`npm run publish:local` will publish all `packages/*` to verdaccio and leave verdaccio running.

# Type of Change

Please Delete options that are not relevant

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Definition of Done

Check all that apply

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Code peer-reviewed.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

**Review by cerebrl — COMMENTED** [review]
It looks like the commit from the token vault PR snuck in here. Did you intend to include it? Otherwise, this looks fine.

**Review by cerebrl — APPROVED** [review]

File: Dockerfile
[the remark this answers is not in the document]
Comment (ryanbas21) [reply Dockerfile:1]: So - as of the state of this repo, the dockerfiles serve no purpose. If we were to deploy e2e's, maybe they could come back for that.

Given the samples here are removed - we would no longer need dockers for samples, and the root Dockerfile was really just a base image for the rest.

File: e2e/autoscript-suites/src/utilities/setup-and-go.ts [hunk 1]
```diff
@@ -72,21 +71,25 @@ export async function setupAndGo(
     console.log(url.toString());
   }
 
-  await page.goto(url.toString(), { waitUntil: 'commit' });
-
   // Listen for events on page
   page.on('console', async (msg) => {
-    return messageArray.push(msg.text());
+    messageArray.push(msg.text());
+    return Promise.resolve(true);
```
[the remark this answers is not in the document]
Comment (ryanbas21) [reply e2e/autoscript-suites/src/utilities/setup-and-go.ts:77]: It isn't i can remove it, i think i was just testing things.

File: .verdaccio/htpasswd
[the remark this answers is not in the document]
Comment (ryanbas21) [reply .verdaccio/htpasswd:1]: These are fake passwords. Nothing real about them. its just for verdaccio to login which itself is a fake proxy. npm, being the joy it is, requires tokens for publish even if its fake.

**Pull request merged**